nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppLibClassLoaderServiceImpl.java

```diff
@@ -422,7 +423,11 @@ private static class Library {
                 }
                 method.setAccessible(true);
                 field.setAccessible(true);
-            } catch (ReflectiveOperationException e) {
+            } catch (ReflectiveOperationException | InaccessibleObjectException e) {
```

Comment:
I think, this catch block is redundant in this case.

Reply:
It’s definitely needed. The setAccessible method throws InaccessibleObjectException if sun.nio.fs is not opened in Java17+. 

And InaccessibleObjectException doesn’t extend ReflectiveOperationException.

Reply:
Both `InaccessibleObjectException` and `ReflectiveOperationException` extends `Exception`. This:

```java
catch (Exception e) {
    LOG.log(WARNING, () -> "Could not initialize sun.nio.fs. We will fallback to basic file attributes. Error: " + e.getMessage(), e);
    method = null;
    field = null;
```

covers all cases and is sufficient. Or not?

Reply:
> However, I'm not sure whether we need to call the sun nio methods? The readAttributes method is called only from the equals method. Is it needed to have all available attributes? Are basic attributes not enough to compare libraries?

Main purpose of sun.nio.* methods - work with descriptor of the open file if possible. This prevents inconsistency if library was replaced between `readAttributes` and snapshot creation.

Reply:
> InaccessibleObjectException happens because of the JPMS, and is kind of expected if add-opens is missing. The add-opens for sun nio can be missing simply because people upgrade to a newer version of GlassFish and keep their domain. I thought it's not appropriate to log the stacktrace.

Perhaps, we should give the user some hint about misconfiguration.

Reply:
OK, I have a new versoin now:
* `InaccessibleObjectException` logs a WARNING with instructions to add --add-opens, no stacktrace - it's clear how to fix this, no need for garbage in logs
* `ReflectiveOperationException` logs a WARNING, no stacktrace, can only happen if the underlying JDK doesn't contain the expected classes - shouldn't happen, if it does, it's clear what is the issue
* another exception - also print stacktrace